src/Routes/Recursion.ts

```diff
@@ -0,0 +1,259 @@
+
+type BlockHeightRequest = FastifyRequest<{
+  Params: {
+    height: number;
+  };
+}>;
+
+type BlockInfoRequest = FastifyRequest<{
+  Params: {
+    query: string;
+  };
+}>;
+
+type InscriptionIdRequest = FastifyRequest<{
+  Params: {
+    inscriptionId: string;
+  };
+}>;
+
+type InscriptionIdPageRequest = FastifyRequest<{
+  Params: {
+    inscriptionId: string;
+    page: number;
+  };
+}>;
+
+type SatNumberRequest = FastifyRequest<{
+  Params: {
+    satNumber: number;
+  };
+}>;
+
+type SatPageRequest = FastifyRequest<{
+  Params: {
+    satNumber: number;
+    page: number;
+  };
+}>;
+
+type SatIndexRequest = FastifyRequest<{
+  Params: {
+    satNumber: number;
+    index: number;
+  };
+}>;
+
+fastify.get(
+  "/r/blockhash/:blockheight",
+  {
+    schema: {
+      params: {
+        height: { type: "number" },
+      },
+    },
+  },
+  async (request: BlockHeightRequest, reply) => {
+    const blockhash = await rpc.blockchain.getBlockHash(request.params.height);
+    reply.send({ blockhash });
+  },
+);
+
+fastify.get("/r/blockhash", async () => {
+  return rpc.blockchain.getLatestBlock().then((block) => block.hash);
+});
+
+fastify.get("/r/blockheight", async () => {
+  return rpc.blockchain.getBlockCount();
+});
+
+fastify.get(
+  "/r/blockinfo/:query",
+  {
+    schema: {
+      params: {
+        query: { type: "string" },
+      },
+    },
+  },
+  async (request: BlockInfoRequest) => {
+    return rpc.blockchain.getBlock(request.params.query, 2);
+  },
+);
+
+fastify.get("/r/blocktime", async () => {
+  return rpc.blockchain.getLatestBlock().then((block) => block.time);
+});
+
+fastify.get(
+  "/r/children/:inscriptionId",
+  {
+    schema: {
+      params: {
+        inscriptionId: { type: "string" },
+      },
+    },
+  },
+  async (request: InscriptionIdRequest) => {
+    const inscription = await db.inscriptions.getInscriptionById(request.params.inscriptionId);
+    if (inscription === undefined) {
+      throw new Error("Inscription not found");
+    }
+    return inscription.children ? inscription.children.slice(0, 100) : [];
+  },
+);
+
+fastify.get(
+  "/r/children/:inscriptionId/:page",
+  {
+    schema: {
+      params: {
+        inscriptionId: { type: "string" },
+        page: { type: "number" },
+      },
+    },
+  },
+  async (request: InscriptionIdPageRequest) => {
+    const inscription = await db.inscriptions.getInscriptionById(request.params.inscriptionId);
+    const pageIndex = request.params.page - 1;
+    if (inscription === undefined) {
+      throw new Error("Inscription not found");
+    }
+    return inscription.children ? inscription.children.slice(pageIndex * 100, (pageIndex + 1) * 100) : [];
+  },
+);
+
+fastify.get(
+  "/r/inscription/:inscriptionId",
+  {
+    schema: {
+      params: {
+        inscriptionId: { type: "string" },
+      },
+    },
+  },
+  async (request: InscriptionIdRequest) => {
+    return db.inscriptions.getInscriptionById(request.params.inscriptionId);
```

Comment:
are the response object from this function exactly equals to ord server response for this recursive endpoint? if not, i think we need to make sure the response are the same

Reply:
as discussed, will be addressing in a separate PR